tests/v2/validation/rbac/rbacuserroles/restrictedadmin_role_test.go

```diff
@@ -0,0 +1,84 @@
+package roles
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters/kubernetesversions"
+	"github.com/rancher/rancher/tests/framework/extensions/provisioning"
+	"github.com/rancher/rancher/tests/framework/extensions/provisioninginput"
+	rbac "github.com/rancher/rancher/tests/framework/extensions/rbac"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type RestrictedAdminTestSuite struct {
+	suite.Suite
+	client  *rancher.Client
+	session *session.Session
+}
+
+func (ra *RestrictedAdminTestSuite) TestRestrictedAdminCreateCluster() {
+	subSession := ra.session.NewSession()
+	defer subSession.Cleanup()
+
+	_, restrictedAdminClient, err := rbac.SetupUser(ra.client, rbac.RestrictedAdmin)
+	require.NoError(ra.T(), err)
+	ra.T().Logf("Validating restricted admin can create an RKE1 cluster")
+	userConfig := new(provisioninginput.Config)
+	config.LoadConfig(provisioninginput.ConfigurationFileKey, userConfig)
+	nodeProviders := userConfig.NodeProviders[0]
+	nodeAndRoles := []provisioninginput.NodePools{
+		provisioninginput.AllRolesNodePool,
+	}
+	externalNodeProvider := provisioning.ExternalNodeProviderSetup(nodeProviders)
+	clusterConfig := clusters.ConvertConfigToClusterConfig(userConfig)
+	clusterConfig.NodePools = nodeAndRoles
+	kubernetesVersion, err := kubernetesversions.Default(restrictedAdminClient, clusters.RKE1ClusterType.String(), []string{})
+	require.NoError(ra.T(), err)
+
+	clusterConfig.KubernetesVersion = kubernetesVersion[0]
+	clusterConfig.CNI = userConfig.CNIs[0]
+	clusterObject, _, err := provisioning.CreateProvisioningRKE1CustomCluster(restrictedAdminClient, &externalNodeProvider, clusterConfig)
+	require.NoError(ra.T(), err)
+	provisioning.VerifyRKE1Cluster(ra.T(), restrictedAdminClient, clusterConfig, clusterObject)
+
```

Comment:
Sorry I did not understanding - What is this test validating? 

provisioning.VerifyRKE1Cluster is taking care of validation also?

Reply:
We are validating cluster creation as a restricted admin.

> provisioning.VerifyRKE1Cluster is taking care of validation also?

yes, validations happen in VerifyRKE1Cluster function.